_posts/2020-04-11-native-mobile-to-flutter.markdown

```diff
@@ -0,0 +1,122 @@
+
+- Project spanned Aug 2019 - Dec 2019 
+- Initial release to Android users in February 2020.
+- Release to all users Android + iOS March 2020.
+- As of the time of this article we've been live across all of our users for about a month with no Flutter related issues.
+
+
+### Why did we do it?
+
+We had two working native mobile applications, but lacked feature parity for Android and iOS users.
+
+Android and iOS development took place at different paces, with varying quality and cost.
+
+Maintaining redundancy across two mobile teams required us to maintain a minimum of four developers on mobile rather than as few as two in a cross platform environment.
+
+### We wanted to achieve...
+
+- feature parity on both platforms.
+- a unified mobile team with higher quality peer reviews. 
+- increased velocity and reduced cost per feature. 
+- higher quality software. 
+- scalability and redundancy of our mobile team.
+
+This seemed possible as the cross-platform mobile space has matured over the last year: there were two great options to choose from, _Flutter_ and _React Native_.
+
+### Why did we choose Flutter? 
+
+The biggest challenge we faced was the timeline within which we had to complete the switch: 5 months.  Our native apps had 140,000 lines of code between the two platforms, so we needed to factor in development speed.
+
+_Regardless of our choice our entire team would have to..._
+
+1. learn a new programming language
+2. learn a new UI framework
+3. add sufficient test coverage to confidently launch the application
+4. delivery a highly secure application. 
+
+Although React Native is a clear winner for a javascript-centric development team, our mobile engineers are not javascript savvy.
+
+#### Here was our decision criteria:
+
+- Community size - _+1 React_
+- Maturity - _+1 React_
+- Learning curve. - _+1 Flutter_
+- Complexity. - _+1 Flutter_
+- Development speed. - _+1 Flutter_
+- Tooling and IDE.  - _+1 Flutter_
+- Testability. - _+1 React_ 
+- Shared code - _+1 Flutter_
+- Platform independence - _+1 Flutter_
+
+
+#### Things we liked about React:
+
+- Maturity and community size: React Native has been in production for much longer.
+- Testing utilities are definitely better on React.
+- More third party library support.
+
+#### Things we liked about Flutter:
+
+- Without javascript developers on our mobile team, the learning curve for Dart/Flutter was substantially lower than Javascript/React Native. 
+- Flutter offered clean, straight-forward conventions that were easy to learn, expressive and explicit.
+- Android Studio offered a turnkey development environment with performance monitoring tools, hot reload and step through debugging and inspection.  Since most of our developers are Android developers, the IDE was an added bonus due to its familiarity.
+- Flutter does not rely on any native UI components, so it's largely immune to breaking changes introduced by Apple, Google, or Android device manufacturers.
```

Comment:
You seem to imply here is that UI stuff breaks more often than not-UI stuff - is that true in your experience?  If so would be nice to explicitly state. 

Also if RN has had a particularly painful upgrade cycle due to this, it'd be great to link as supporting evidence (maybe android 10?). 

Finally, it seems like there's lots of "not-UI" stuff, like biometrics, gps, camera APIs etc. that can break too. And because cross-platform apps of a certain complexity seem to almost always bind to some of these native features - it'd seem like most cross-platform apps would be vulnerable to these kinds of changes, regardless of the framework chosen.  That's really more of an "app problem" than a "framework problem" though.